memiavl/db.go

```diff
@@ -142,29 +168,44 @@ func (db *DB) ApplyUpgrades(upgrades []*TreeNameUpgrade) error {
 	return nil
 }
 
-// cleanupSnapshotRewrite cleans up the old snapshots and switches to a new multitree
-// if a snapshot rewrite is in progress. It returns true if a snapshot rewrite has completed
-// and false otherwise, along with any error encountered during the cleanup process.
-func (db *DB) cleanupSnapshotRewrite() (bool, error) {
-	if db.snapshotRewriteChan == nil {
-		return false, nil
+// checkAsyncTasks checks the status of background tasks non-blocking-ly and process the result
+func (db *DB) checkAsyncTasks() error {
+	return errors.Join(
+		db.checkAsyncWAL(),
+		db.checkBackgroundSnapshotRewrite(),
+	)
+}
+
+// checkAsyncWAL check the quit signal of async wal writing
+func (db *DB) checkAsyncWAL() error {
+	select {
+	case err := <-db.walQuit:
+		// async wal writing failed, we need to abort the state machine
+		return fmt.Errorf("async wal writing goroutine quit unexpectedly: %w", err)
```

Comment:
need handle nil here
```suggestion
		if err != nil {
			// async WAL writing failed, we need to abort the state machine
			return fmt.Errorf("async WAL writing goroutine quit unexpectedly: %w", err)
		}
```

Reply:
it also treat `nil` as an error here, the async wal goroutine just not expected to quit.

Reply:
close will return right?

Reply:
yes, but whenever it quit, the state machine should be aborted